File: ChineseAuction.Api.Angular/client/ChineseAuctionAngular/src/app/models/DonorDTO.ts
import { Gift } from './GiftDTO';

export class DonorDTO {
  idDonor: number = 0;
  firstName: string = '';
  lastName: string = '';
  email: string = '';
  phoneNumber: string = '';

  constructor(init?: Partial<DonorDTO>) {
    Object.assign(this, init);
  }
}

export class DonorCreateDTO {
  firstName: string = '';
  lastName: string = '';
  email: string = '';
  phoneNumber: string = '';

  constructor(init?: Partial<DonorCreateDTO>) {
    Object.assign(this, init);
  }
}

export class Donor {
  idDonor: number = 0;
  firstName: string = '';
  lastName: string = '';
  email: string = '';
  phoneNumber: string = '';
  gifts?: Gift[] = []; 

  constructor(init?: Partial<Donor>) {
    Object.assign(this, init);
  }
}

File: ChineseAuction.Api.Angular/server/ChineseAuctionAPI/DTOs/DonorDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ChineseAuctionAPI.DTOs
{
    public class DonorDTO
    {
        public int IdDonor { get; set; }
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [EmailAddress, Required]
        public string Email { get; set; }
        [Phone, Required]
        public string PhoneNumber { get; set; }
    }

    public class DonorCreateDTO
    {
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [EmailAddress]
        public string Email { get; set; }
        [Phone]
        public string PhoneNumber { get; set; }
    }
}